File: Calculator/app/src/main/java/kg/jedi/calculator/MainActivity.java
package kg.jedi.calculator;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private TextView result;
    private EditText number1, number2;
    private Button add, divide, subtract, multiply;

    private EventListener eventListener = new EventListener();


    float calcResult, num;

    float num1, num2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        init();
        setListeners();
    }

    private void init() {
        result = (TextView)findViewById(R.id.result);

        number1 = (EditText)findViewById(R.id.number1);
        number2 = (EditText)findViewById(R.id.number2);

        add = (Button)findViewById(R.id.add);
        divide = (Button)findViewById(R.id.divide);
        subtract = (Button)findViewById(R.id.subtract);
        multiply = (Button)findViewById(R.id.multiply);
    }

    private void setListeners() {
        add.setOnClickListener(eventListener);
        divide.setOnClickListener(eventListener);
        subtract.setOnClickListener(eventListener);
        multiply.setOnClickListener(eventListener);
    }

    private class EventListener implements View.OnClickListener {

        @Override
        public void onClick(View view) {

            num1 = Integer.valueOf(number1.getText().toString());
            num2 = Integer.valueOf(number2.getText().toString());

            if (view.equals(add)) {
                calcResult = (num1) + (num2);
            }
            if (view.equals(divide)) {
                calcResult = (num1) / (num2);
            }
            if (view.equals(multiply)) {
                calcResult = (num1) * (num2);
            }
            if (view.equals(subtract)) {
                calcResult = (num1) - (num2);
            }

            result.setText(String.valueOf(calcResult));
        }
    }
}

File: ToDoList/app/src/main/java/kg/jedi/todolist/MainActivity.kt
package kg.jedi.todolist

import android.content.Intent
import android.database.Cursor
import android.os.Bundle
import android.support.v4.app.LoaderManager
import android.support.v4.content.AsyncTaskLoader
import android.support.v4.content.Loader
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.support.v7.widget.helper.ItemTouchHelper
import android.util.Log
import android.widget.Toast
import kg.jedi.todolist.data.TaskContract
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity(), LoaderManager.LoaderCallbacks<Cursor> {

    private lateinit var mAdapter: CustomCursorAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        mAdapter = CustomCursorAdapter(this)

        recyclerViewTasks.adapter = mAdapter
        recyclerViewTasks.layoutManager = LinearLayoutManager(this)
        setSwipeListeners(recyclerViewTasks)

        btnAddTask.setOnClickListener({
            val addTaskIntent = Intent(this@MainActivity, AddTaskActivity::class.java)
            startActivity(addTaskIntent)
        })

        supportLoaderManager.initLoader(TASK_LOADER_ID, null, this)
    }

    private fun setSwipeListeners(recyclerView: RecyclerView) {

        ItemTouchHelper(object : ItemTouchHelper.SimpleCallback(0,
                ItemTouchHelper.LEFT or ItemTouchHelper.RIGHT) {

            override fun onMove(recyclerView: RecyclerView, viewHolder: RecyclerView.ViewHolder,
                                target: RecyclerView.ViewHolder): Boolean {
                return false
            }

            override fun onSwiped(viewHolder: RecyclerView.ViewHolder, swipeDir: Int) {
                val id = viewHolder.itemView.tag as Int
                val stringId = Integer.toString(id)

                var uri = TaskContract.TaskEntry.CONTENT_URI
                uri = uri.buildUpon().appendPath(stringId).build()

                contentResolver.delete(uri, null, null)
                Toast.makeText(baseContext, "Task has deleted!", Toast.LENGTH_SHORT).show()
                supportLoaderManager.restartLoader(TASK_LOADER_ID, null, this@MainActivity)
            }
        }).attachToRecyclerView(recyclerView)
    }

    override fun onResume() {
        super.onResume()
        supportLoaderManager.restartLoader(TASK_LOADER_ID, null, this)
    }

    override fun onCreateLoader(id: Int, loaderArgs: Bundle?): Loader<Cursor> {
        return object : AsyncTaskLoader<Cursor>(this) {

            internal var mTaskData: Cursor? = null

            override fun onStartLoading() {
                if (mTaskData != null) {
                    deliverResult(mTaskData)
                } else {
                    forceLoad()
                }
            }

            override fun loadInBackground(): Cursor? {
                try {
                    return contentResolver.query(TaskContract.TaskEntry.CONTENT_URI,
                            null, null, null,
                            TaskContract.TaskEntry.COLUMN_PRIORITY)
                } catch (e: Exception) {
                    Log.e(TAG, "Failed to asynchronously load data.")
                    e.printStackTrace()
                    return null
                }
            }

            override fun deliverResult(data: Cursor?) {
                mTaskData = data
                super.deliverResult(data)
            }
        }
    }

    override fun onLoadFinished(loader: Loader<Cursor>, data: Cursor) {
        mAdapter.swapCursor(data)
    }

    override fun onLoaderReset(loader: Loader<Cursor>) {
        mAdapter.swapCursor(null)
    }

    companion object {
        private val TAG = MainActivity::class.java.simpleName
        private val TASK_LOADER_ID = 0
    }

}

